### PR TITLE
Resolve cross configure issues for the tools

### PR DESCRIPTION
**Pull request by charliegracie**
In PR #383 the tools configuration was significantly improved but
it removed the ability for the tools to be cross configured.  If the
top level configure script can be cross configured the tool script also
requires that ability. This change allows the tools to be cross
configured as long as you specify all of the compile options by hand.

This may not be the ideal end solution but it resolves issues with
runtimes that cross configure OMR. If the tools configure script 
was able to just use the values created by the top level configure
script and not perform any checks that may be a better solution.

Signed-off-by: Charlie Gracie <charlie.gracie@gmail.com>

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -859,7 +857,7 @@ enable_OMR_THR_FORK_SUPPORT
 enable_OMR_THR_THREE_TIER_LOCKING
 enable_OMR_THR_CUSTOM_SPIN_OPTIONS
 enable_OMR_NOTIFY_POLICY_CONTROL
-enable_OMR_THR_SYNCHRONIZATION_PROTOTYPES
+enable_OMR_THR_SPIN_WAKE_CONTROL
```
Comment (mstoodle) [inline configure]: Is this an extra you're throwing into the bargain? :)

**Comment by mstoodle** [comment]
Hi @charliegracie ,

Slightly confused because I haven't been tracking this work before, so trying to understand the scenario here: we need to be able to build the tools on a platform (the BUILD system?) that's different than the one we're configuring OMR for (the HOST system)? Or is OMR being cross configured for a build system different than the current host and this change simply allows the tools to also be cross configured for that build system?

**Comment by charliegracie** [comment]
@mstoodle it is the second one.  OMR including the tools are being configured on a host system for a different build system.  An example would be running configure on Linux x86 and the compiling on AIX.  This is not a common way of using configure but since the main level configure script in OMR supports it I think the tool one should also support it since it is recursively launched via the main configure.

**Comment by mstoodle** [comment]
@charliegracie Given that our tests won't be testing this change, has it been tested inside IBM (not that we should be relying on that testing, but having that testing and knowing it works is better than not knowing those things). A whole cross-compile build in OMR feels like overkill for this kind of thing; I just want to document that it has been tested and known to work for the scenario it's designed for.

**Comment by charliegracie** [comment]
@mstoodle yes and it is currently required for some of the languages consuming OMR and they have been unable to accept any new OMR changes until this is resolved.

**Comment by mstoodle** [comment]
ok, thanks Charlie. Looks good!

**Pull request merged**